### PR TITLE
Telemetry Integration

### PR DESCRIPTION
**Pull request by heidijinxujia**
Wire up the telemetry configuration with json.
- json file
- initilization

TODO
Add events into the calls.

### CODE REVIEW

File: msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java [hunk 1]
```diff
@@ -380,6 +383,26 @@ protected PublicClientApplication(@NonNull final Context context,
...
 
+    private void setupTelemetry(@NonNull final Context context,
+                                @Nullable final PublicClientApplicationConfiguration developerConfig) {
+        if (null != developerConfig.getTelemetryConfiguration()) {
+            com.microsoft.identity.common.internal.logging.Logger.verbose(
+                    TAG,
+                    "Telemetry configuration is set. Telemetry is enabled."
+            );
+        } else {
+            com.microsoft.identity.common.internal.logging.Logger.verbose(
```
Comment (rpdome) [inline msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java]: what would happen if we're calling the constructor above with configuration as null?

File: msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java [hunk 2]
```diff
@@ -380,6 +383,26 @@ protected PublicClientApplication(@NonNull final Context context,
         }
     }
 
+    private void setupTelemetry(@NonNull final Context context,
+                                @Nullable final PublicClientApplicationConfiguration developerConfig) {
+        if (null != developerConfig.getTelemetryConfiguration()) {
```
Comment (kreedula) [inline msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java]: `developerConfig` is marked `Nullable` , needs null check for it as well.

File: msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java [hunk 3]
```diff
@@ -380,6 +383,26 @@ protected PublicClientApplication(@NonNull final Context context,
...
+                    "Telemetry configuration is set. Telemetry is enabled."
+            );
+        } else {
+            com.microsoft.identity.common.internal.logging.Logger.verbose(
+                    TAG,
+                    "Telemetry configuration is null. Telemetry is disabled."
+            );
+        }
+
+        new com.microsoft.identity.common.internal.telemetry.Telemetry.Builder()
```
Comment (kreedula) [inline msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java]: shouldn't this be inside the if loop above where we check for `null != developerConfig.getTelemetryConfiguration()`

File: msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java (hunk 1 above)
Comment (heidijinxujia) [reply msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java]: If the telemetry configuration is null, then the Telemetry.isTelemetryEnabled will be set to "false", meaning that the Telemetry is disabled for this client.

File: msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java (hunk 3 above)
Comment (heidijinxujia) [reply msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java]: developerConfig.getTelemetryConfiguration() is a nullable value here. If the developerConfig.getTelemetryConfiguration() is null means that the client does not set up the Telemetry configuration. In this case, the Telemetry will be disabled for this client.

File: msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java [hunk 4]
```diff
@@ -628,6 +628,9 @@ public void onError(MsalException exception) {
     protected PublicClientApplication(@NonNull final Context context,
                                       @Nullable final PublicClientApplicationConfiguration developerConfig) {
         setupConfiguration(context, developerConfig);
+
+        setupTelemetry(context, developerConfig);
```
Comment (heidijinxujia) [inline msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java]: Put it into setupConfiguration and rename `setupConfiguration `

File: msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java (hunk 2 above)
Comment (heidijinxujia) [reply msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java]: Changed form @Nullable to @NonNull. Moved to into setupConfiguration and using mPublicClientConfiguration which is always nonnull.

**Comment by heidijinxujia** [comment]
@kreedula Hey Krishna, let me know if you have any other reviews. Thanks!

**Pull request merged**